stl/inc/flat_set

```diff
@@ -423,64 +446,112 @@ private:
         return true;
     }
 
+    bool _Check_where(const const_iterator _Where, const _Kty& _Val) const {
+        // check that _Val can be inserted before _Where
+        const key_compare& _Compare = _Get_comp();
+        if constexpr (_Multi) {
+            // check that _Where is the upper_bound for _Val
+            // equivalent to checking *(_Where-1) <= _Val < *_Where
+            return (_Where == cend() || _Compare(_Val, *_Where))
+                && (_Where == cbegin() || !_Compare(_Val, *(_Where - 1)));
+        } else {
+            // check that _Where is the lower_bound for _Val, and *_Where is not equivalent to _Val
+            // equivalent to checking *(_Where-1) < _Val < *_Where
+            return (_Where == cend() || _Compare(_Val, *_Where))
+                && (_Where == cbegin() || _Compare(*(_Where - 1), _Val));
+        }
+    }
+
     template <class _Ty>
-        requires (_Keylt_transparent && is_constructible_v<_Kty, _Ty>) || is_same_v<_Ty, _Kty>
-    void _Emplace_hint(const_iterator _Where, _Ty&& _Val) {
-        _Container& _Cont     = _Get_cont();
-        _Keylt& _Compare      = _Get_comp();
-        const iterator _Begin = begin();
-        const iterator _End   = end();
+    auto _Emplace(_Ty&& _Val) {

```

Comment:
From the call sites it might be better to name as `_Insert[_hint]`, as the function does single-element insertion. (though `_Emplace[_hint]` might be ok as the function uses `container.emplace` for performance guarantee...)